src/js/providers/default.ts

```diff
@@ -1,9 +1,68 @@
 import { PLAYER_STATE, MEDIA_TYPE } from 'events/events';
+import { GenericObject, SourceObj, SeekRange } from 'types/generic.type';
+
+const noop: () => void = function(): void { /* noop */ };
+const returnFalse: () => boolean = (() => false);
+const getNameResult: { name: string } = { name: 'default' };
+const returnName: () => { name: string } = (() => getNameResult);
+interface DefaultProvider {
+    state?: string;
+    supports: () => boolean;
+    play: () => void;
+    pause: () => void;
+    preload: () => void;
+    load: () => void;
+    stop: () => void;
+    volume: () => void;
+    mute: () => void;
+    seek: () => void;
+    resize: () => void;
+    remove: () => void; // removes from page
+    destroy: () => void; // frees memory
+
+    setVisibility: () => void;
+    setFullscreen: () => void;
+    getFullscreen: () => boolean;
+    supportsFullscreen: () => boolean;
+
+    // If setContainer has been set; this returns the element.
+    //  It's value is used to determine if we should remove the <video> element when setting a new provider.
+    getContainer: () => void;
+
+    // Sets the parent element; causing provider to append <video> into it
+    setContainer: () => void;
+
+    getName: () => { name: string };
+
+    getQualityLevels: () => void;
+    getCurrentQuality: () => void;
+    setCurrentQuality: () => void;
+
+    getAudioTracks: () => void;
+
+    getCurrentAudioTrack: () => void;
+    setCurrentAudioTrack: () => void;
+
+    getSeekRange: () => SeekRange;
+
+    setPlaybackRate: () => void;
+    getPlaybackRate: () => number;
+    getBandwidthEstimate: () => number | null;
+    getLiveLatency: () => number | null;
+    setControls: () => void;
 
-const noop = function() {};
-const returnFalse = (() => false);
-const getNameResult = { name: 'default' };
-const returnName = (() => getNameResult);
+    attachMedia: () => void;
+    detachMedia: () => void;
+    init: () => void;
+
+    setState: (state: string) => void;
+
+    sendMediaType: (sources: Array<SourceObj>) => void;
+
+    getDuration: () => number;
+
+    trigger: (evt: string, obj: GenericObject) => void;
```

Comment:
This interface should extend the Events interface. You don't need to redefine trigger, on etc..